vm/devices/user_driver/vfio_sys/src/lib.rs

```diff
@@ -338,6 +355,36 @@ impl Device {
         Ok(MappedRegion { addr, len })
     }
 
+    /// Attempts to map the region to the specific VA.
+    pub fn map_to(
```

Comment:
this cannot be a safe function. because you can pass any `addr`, this function must be unsafe, with a contract that documents that the caller needs to pick a valid addr (you'll need to expand on what "valid" means in this particular domain)

Reply:
I'm in the process of removing VA references, so there will be only PFNs/GPNs left. I suppose all of them should be marked as unsafe too? Or safety is guaranteed if we can check that PFN belongs to the reserved region?

Reply:
if you do validation within the function to guarantee the `addr` lies within a valid region, and wouldn't cause any downstream memory safety issues, then the functions can be marked safe. i.e: there should be no combo of input params that could cause memory safety issues

if you're relying on any form of caller validation / guarantees regarding any of the provided params, those validations need to be documented in the function's SAFETY contract, and the function should be marked as unsafe